nautobot_device_lifecycle_mgmt/tests/test_api.py

```diff
@@ -378,6 +391,12 @@ def test_bulk_delete_objects(self):
     def test_bulk_update_objects(self):
         """Currently don't support bulk operations."""
 
+    def test_notes_url_on_object(self):
+        """Currently don't support notes."""
+
+    def test_list_objects_brief(self):
```

Comment:
If the serializer uses NautobotModelSerializer, these list_objects_brief tests should pass and shouldn't need to be disabled. That may be all that's needed to get notes_url working as well.

Reply:
For Notes tests we'll need to implement URLs which I want to do in a separate PR.

Problem with `test_list_objects_brief` is that Nautobot 1.4.0 seems to produce two new fields `created` and `last_updated`. If I allow this test to run we get failure in either Nautobot 1.4.0 (fields missing) or in Nautobot < 1.4.0 (extra fields present).

```
======================================================================
FAIL: test_list_objects_brief (nautobot_device_lifecycle_mgmt.tests.test_api.ValidatedSoftwareLCMAPITest)
GET a list of objects using the "brief" parameter.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/test/utils.py", line 387, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/nautobot/utilities/testing/api.py", line 239, in test_list_objects_brief
    self.assertEqual(sorted(response.data["results"][0]), self.brief_fields)
AssertionError: Lists differ: ['created', 'custom_fields', 'device_roles',[154 chars]lid'] != ['custom_fields', 'device_roles', 'device_ty[127 chars]lid']

First differing element 0:
'created'
'custom_fields'

First list contains 2 additional elements.
First extra element 15:
'url'

- ['created',
-  'custom_fields',
? ^

+ ['custom_fields',
? ^

   'device_roles',
   'device_types',
   'devices',
   'display',
   'end',
   'id',
   'inventory_items',
-  'last_updated',
   'object_tags',
   'preferred',
   'software',
   'start',
   'tags',
   'url',
   'valid']

----------------------------------------------------------------------
Ran 468 tests in 42.913s
```

Reply:
Hmmm.. something's not right here, as the `created` and `last_updated` fields should *not* be present on the nested/brief serializer. Looks like maybe the issue is that you don't have a NestedHardwareLCMSerializer, NestedContactLCMSerializer, or NestedValidatedSoftwareLCMSerializer in the plugin, so the "brief" version of these APIs is actually the same as the full version?

Reply:
Indeed, we don't have nested serializer classes for these three serializers. I'm just not sure what changed that tests pass in 1.3.9 without overriding `test_list_objects_brief` but in 1.4.0 I get test failures due to the additional fields present.

I might just override these test methods for the three classes for now and see if this gets fixed in a separate PR when working adding `NautobotModelSerializer` and notes support.